xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp

```diff
@@ -452,20 +507,177 @@ bool CGUIDialogVideoManagerVersions::ProcessVideoVersion(VideoDbContentType item
     return false;
   }
 
-  for (int i = 0; i < list.Size(); ++i)
+  if (!PostProcessList(list, dbId))
+    return false;
+
+  return ChooseVideoAndConvertToVideoVersion(list, itemType, mediaType, dbId, videodb,
+                                             MediaRole::NewVersion);
+}
+
+bool CGUIDialogVideoManagerVersions::AddVideoVersionFilePicker()
+{
+  const int dbId{m_videoAsset->GetVideoInfoTag()->m_iDbId};
+  const MediaType mediaType{m_videoAsset->GetVideoInfoTag()->m_type};
+  const VideoDbContentType itemType{m_videoAsset->GetVideoContentType()};
+
+  // prompt to choose a video file
+  VECSOURCES sources{*CMediaSourceSettings::GetInstance().GetSources("files")};
+
+  CServiceBroker::GetMediaManager().GetLocalDrives(sources);
+  CServiceBroker::GetMediaManager().GetNetworkLocations(sources);
+
+  std::string path;
+  if (CGUIDialogFileBrowser::ShowAndGetFile(
+          sources, CServiceBroker::GetFileExtensionProvider().GetVideoExtensions(),
+          g_localizeStrings.Get(40014), path))
   {
-    if (dbId == list[i]->GetVideoInfoTag()->m_iDbId)
+    std::string typeVideoVersion;
+    std::string videoTitle;
+    int idFile{-1};
+    int idMedia{-1};
+    MediaType itemMediaType;
+    VideoAssetType videoAssetType{VideoAssetType::UNKNOWN};
+
+    const int idVideoVersion{m_database.GetVideoVersionInfo(path, idFile, typeVideoVersion, idMedia,
+                                                            itemMediaType, videoAssetType)};
+
+    if (idVideoVersion != -1)
     {
-      list.Remove(i);
-      break;
+      CFileItemList versions;
+      m_database.GetVideoVersions(itemType, dbId, versions);
+      if (std::any_of(versions.begin(), versions.end(),
+                      [idFile](const std::shared_ptr<CFileItem>& version)
+                      { return version->GetVideoInfoTag()->m_iDbId == idFile; }))
```

Comment:
Could you please add a `@todo` here. It seems we're obtaining all versions just to look for one specific item, then. As far as I can see, `versions` is not used elsewhere in this method.

Reply:
ok